api/lib/opentelemetry/context/propagation/rack_env_getter.rb

```diff
@@ -7,6 +7,9 @@
 module OpenTelemetry
   class Context
     module Propagation
+      # @deprecated Use the rack env getter found in the
```

Comment:
We should add a removal date to this comment. 🤔 

[API support guarantees](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#api-support) are more complex than I thought.

> Major versions of the API MUST be supported for a minimum of three years after the release of the next major API version.

Technically, this means we can simply remove the `RackEnvGetter` and bump the major version, but we then need to support both the latest major version and the major version at the time of writing for 3 years from the date of the major version bump.

So maybe the approach for now is:
1. Add the deprecation notice without a removal date.
2. Figure out and document the support process for old major versions (e.g. how to PR against older versions, how CI and releases work).
3. Remove the deprecated class and bump the major version.